sdk/core/Azure.Core/src/RequestFailedException.cs

```diff
@@ -2,51 +2,94 @@
 // Licensed under the MIT License.
 
 using System;
+using System.ClientModel;
 using System.Collections.Generic;
 using System.ComponentModel;
 using System.Globalization;
-using System.IO;
 using System.Runtime.Serialization;
 using System.Text;
+using System.Threading.Tasks;
 using Azure.Core;
 using Azure.Core.Pipeline;
 
 namespace Azure
 {
-#pragma warning disable CA2229, CA2235 // False positive
     /// <summary>
     /// An exception thrown when service request fails.
     /// </summary>
     [Serializable]
-    public class RequestFailedException : Exception, ISerializable
+    public class RequestFailedException : ClientResultException, ISerializable
     {
         private const string DefaultMessage = "Service request failed.";
 
         /// <summary>
-        /// Gets the HTTP status code of the response. Returns. <code>0</code> if response was not received.
+        /// TBD.
         /// </summary>
-        public int Status { get; }
+        /// <param name="response"></param>
+        /// <param name="parser"></param>
+        /// <param name="innerException"></param>
+        /// <returns></returns>
+        public static async ValueTask<RequestFailedException> CreateAsync(Response response, RequestFailedDetailsParser? parser = default, Exception? innerException = default)
+        {
+            ErrorDetails details = await CreateExceptionDetailsAsync(response, parser).ConfigureAwait(false);
+            return new RequestFailedException(response, details, innerException);
+        }
 
         /// <summary>
         /// Gets the service specific error code if available. Please refer to the client documentation for the list of supported error codes.
         /// </summary>
         public string? ErrorCode { get; }
 
-        /// <summary>
-        /// Gets the response, if any, that led to the exception.
-        /// </summary>
-        private readonly Response? _response;
+        #region Response constructors
```

Comment:
Why do you hate us all?

Reply:
Do you not like `#region`?  OMG, @jsquire, I had no idea!

In all honesty, the fact that we have eleven constructors for this type kind of makes me want to cry.  Somewhere in a OneNote file I have a diagram where they're numbered and color coded because it was the only way I could understand what was happening here.  But if you hate `#region` I'll pull it out and just try to add a helpful comment for the next person that comes through.